source/common/midi_creation.py

```diff
@@ -1,37 +1,295 @@
+
+        #add piano notes
+        if amount_of_piano > 0 and shape.instrument == "piano":
+            
+            notes = piano_notes(scale(shape.pitch, "piano"))
+
+            # add an accord
+            midi_piano.addNote(0, 0, notes[0], shape.note_placement, 1, shape.volume) # add a note
+            midi_piano.addNote(0, 0, notes[1], shape.note_placement, 1, shape.volume) # add a note
+            midi_piano.addNote(0, 0, notes[2], shape.note_placement, 1, shape.volume) # add a note
+
+            object_piano += 1
+
+        #add drum notes
+        if amount_of_drum > 0 and shape.instrument == "drum":
+            
+            midi_drum.addNote(0, 0, 84, shape.note_placement, 1, shape.volume) # add a note
+
+            object_drum +=1
+
+        #add saxophone notes
+        if amount_of_saxophone > 0 and shape.instrument == "saxophone":
+            
+            midi_saxophone.addNote(0, 0, scale(shape.pitch, "saxophone"), shape.note_placement, 1, shape.volume) # add a note
+
+            object_saxophone +=1
+
+        #add kick notes
+        if amount_of_kick > 0 and shape.instrument == "kick":
+            
+            midi_kick.addNote(0, 0, 84, shape.note_placement, 1, shape.volume) # add a note
+
+            object_kick +=1
+        
+        #add clap notes
+        if amount_of_clap > 0 and shape.instrument == "clap":
+        
+            midi_clap.addNote(object_clap, channel_clap, 84, shape.note_placement, 1, shape.volume) # add a note
+
+            object_clap +=1
         
-        pygame.mixer.init()
-        #load MIDI file
-        pygame.mixer.music.load("output.mid")
-        # play MIDI file
-        pygame.mixer.music.play()
-        # wait for music to finish playing
-        while pygame.mixer.music.get_busy():
-            continue
\ No newline at end of file
+        #Mmake wav files of all the midi's when al nodes are added
+        if amount_of_instruments -1 == iteration:
+            if amount_of_flute == 0:
+                midi_flute.addNote(0, 0, 0, 1, 1, 120) #clear track
+            if amount_of_violin == 0:
+                midi_violin.addNote(0, 0, 0, 1, 1, 120) #clear track 
+            if amount_of_guitar == 0:
+                midi_guitar.addNote(0, 0, 0, 1, 1, 120) #clear track
+            if amount_of_piano == 0:
+                midi_piano.addNote(0, 0, 0, 1, 1, 120) #clear track
+            if amount_of_drum == 0:
+                midi_drum.addNote(0, 0, 0, 1, 1, 120) #clear track 
+            if amount_of_saxophone == 0:
+                midi_saxophone.addNote(0, 0, 0, 1, 1, 120) #clear track
+            if amount_of_kick == 0:
+                midi_kick.addNote(0, 0, 0, 1, 1, 120) #clear track
+            if amount_of_clap == 0:
+                midi_clap.addNote(0, 0, 0, 1, 1, 120) #clear track
+
+            #write all the midi files
+            with open(model_custom_path + "\\flute_output.mid", "wb") as output_file1: 
+                midi_flute.writeFile(output_file1)
```

Comment:
delete \\. use os.path instead

Reply:
model_custom_path = os.path.join(os.getcwd(), 'files', 'audio_generator', 'midi_files') #path of the audio files save location

create custom path for every output